### PR TITLE
fix(asm): workaround disabling builtins.open aspects

### PR DESCRIPTION
**Pull request by juanjux**
## Description

We found issues dogfooding on dogweb stagging with one specific test that used the `open` aspect for path traversal. This happens in 2.8.4 too but not in 2.7.10. To discard that the problem is IAST-related (or fix the issue if it is), this PR disables the patching of `unittest.mock` and the `builtins.open` aspect.

## Checklist

- [X] Change(s) are motivated and described in the PR description
- [X] Testing strategy is described if automated tests are not included in the PR
- [X] Risks are described (performance impact, potential for breakage, maintainability)
- [X] Change is maintainable (easy to change, telemetry, documentation)
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [X] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by gnufede — APPROVED** [review]

**Comment by brettlangdon** [comment]
wrong PR base?

**Comment by juanjux** [comment]
> wrong PR base?

It's a workaround to unblock the 2.9 release and we will investigate from main tomorrow. Merging into main would make some system tests fail (for everybody) so I set 2.9 as base.

File: ddtrace/appsec/_iast/_taint_tracking/Aspects/AspectsOsPath.cpp [hunk 1]
```diff
@@ -108,7 +108,7 @@ api_ospathbasename_aspect(const StrType& path)
     auto basename_result = basename(path);
 
     auto tx_map = initializer->get_tainting_map();
-    if (not tx_map or py::len(basename_result) == 0) {
+    if (not tx_map or tx_map->empty() or py::len(basename_result) === 0) {
```
Comment (avara1986) [inline ddtrace/appsec/_iast/_taint_tracking/Aspects/AspectsOsPath.cpp:111]:
```suggestion
    if (not tx_map or tx_map->empty() or py::len(basename_result) == 0) {
```

File: pyproject.toml [hunk 2]
```diff
@@ -68,7 +68,7 @@ Homepage = "https://github.com/DataDog/dd-trace-py"
 "Source Code" = "https://github.com/DataDog/dd-trace-py/"
 
 [tool.setuptools_scm]
-version_scheme = "release-branch-semver"  # Must be "release-branch-semver" for now in main, see https://github.com/DataDog/dd-trace-py/issues/8801
+version_scheme = "guess-next-dev"
```
Comment (avara1986) [inline pyproject.toml:71]:
```suggestion
version_scheme = "release-branch-semver"  # Must be "release-branch-semver" for now in main, see https://github.com/DataDog/dd-trace-py/issues/8801
```
revert

**Comment by avara1986** [comment]
> wrong PR base?

Fix the PR base @brettlangdon

**Comment by juanjux** [comment]
System tests updated so no need to rebase to 2.9 anymore @brettlangdon

**Review by brettlangdon — APPROVED** [review]

**Review by avara1986 — APPROVED** [review]

**Pull request merged**